packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+import {FsxFileSystemAttributes, FsxFileSystemBase, FsxFileSystemProps, IFsxFileSystem} from './file-system';
+import {CfnFileSystem} from './fsx.generated';
+
+/**
+ * The different kinds of file system deployments used by Lustre.
+ */
+export enum FsxLustreDeploymentType {
+  /**
+   * Original type for shorter term data processing. Data is not replicated and does not persist on server fail.
+   */
+  SCRATCH_1 = 'SCRATCH_1',
+  /**
+   * Newer type for shorter term data processing. Data is not replicated and does not persist on server fail.
+   * Provides better support for spiky workloads.
+   */
+  SCRATCH_2 = 'SCRATCH_2',
+  /**
+   * Long term storage. Data is replicated and file servers are replaced if they fail.
+   */
+  PERSISTENT_1 = 'PERSISTENT_1'
+}
+
+/**
+ * The configuration for the Amazon FSx for Lustre file system.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-lustreconfiguration.html
+ */
+export interface LustreConfiguration {
+  /**
+   * The type of backing file system deployment used by FSx.
+   */
+  readonly deploymentType: FsxLustreDeploymentType;
+
+  /**
+   * The path in Amazon S3 where the root of your Amazon FSx file system is exported. The path must use the same
+   * Amazon S3 bucket as specified in ImportPath. If you only specify a bucket name, such as s3://import-bucket, you
+   * get a 1:1 mapping of file system objects to S3 bucket objects. This mapping means that the input data in S3 is
+   * overwritten on export. If you provide a custom prefix in the export path, such as
+   * s3://import-bucket/[custom-optional-prefix], Amazon FSx exports the contents of your file system to that export
+   * prefix in the Amazon S3 bucket.
+   *
+   * @default s3://import-bucket/FSxLustre[creation-timestamp]
+   */
+  readonly exportPath?: string;
+
+  /**
+   * For files imported from a data repository, this value determines the stripe count and maximum amount of data per
+   * file (in MiB) stored on a single physical disk. Allowed values are between 1 and 512,000.
+   *
+   * @default 1024
+   */
+  readonly importedFileChunkSize?: number;
+
+  /**
+   * The path to the Amazon S3 bucket (including the optional prefix) that you're using as the data repository for
+   * your Amazon FSx for Lustre file system. Must be of the format "s3://{bucketName}/optional-prefix" and cannot
+   * exceed 900 characters.
+   *
+   * @default - no bucket is imported
+   */
+  readonly importPath?: string;
+
+  /**
+   * Required for the PERSISTENT_1 deployment type, describes the amount of read and write throughput for each 1
+   * tebibyte of storage, in MB/s/TiB. Valid values are 50, 100, 200.
+   *
+   * @default - no default, conditionally required for PERSISTENT_1 deployment type
+   */
+  readonly perUnitStorageThroughput?: number;
+
+  /**
+   * The preferred time to perform weekly maintenance, in the UTC time zone, 24 hour clock. For example: '2:20:30'.
```

Comment:
How is it we have *weekly* maintenance but don't specify a day?

Reply:
the first digit is the day of the week, say 2:20:30 means Tuesday at 20:30

Reply:
That is a super non-obvious encoding (to me).

I'd like some syntactic sugar over that here, probably looking something like:

```ts
  weeklyMaintenanceStartTime: MaintenanceTime.utc(Weekday.TUESDAY, '20:30')
// or
  weeklyMaintenanceStartTime: MaintenanceTime.utc(Weekday.TUESDAY, 20, 30)

```

(I know that's a mouthful, I'd be open to other suggestions as long as we can cleanly separate the different fields in a readable way).

Reply:
Makes sense to me.

Reply:
I took a stab at it, let me know what you think.